src/main/java/de/tum/in/www1/artemis/service/plagiarism/ModelingPlagiarismDetectionService.java

```
@@ -46,6 +50,12 @@ public ModelingPlagiarismDetectionService(PlagiarismWebsocketService plagiarismW
      */
     public ModelingPlagiarismResult checkPlagiarism(ModelingExercise exerciseWithParticipationsSubmissionsResults, double minimumSimilarity, int minimumModelSize,
             int minimumScore) {
+        var courseId = exerciseWithParticipationsSubmissionsResults.getCourseViaExerciseGroupOrCourseMember().getId();
+        if (plagiarismCacheService.isActivePlagiarismCheck(courseId)) {
+            return null;
```

we should throw an exception here